src/Database/Auth/ApiKeyAuthenticator.ts

```diff
@@ -101,14 +102,14 @@ export class ApiKeyAuthenticator {
       let publicKey: Buffer, body: IResponseBody;
 
       if (!StatusCodes.isOk(response.statusCode)
-        && !TypeUtil.isObject(body = response.body) && !body.PublicKey
+        && !TypeUtil.isObject(body = response.body as IResponseBody) && !body.PublicKey
```

Comment:
It's not the case in C#:

https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Documents/Operations/Indexes/DeleteIndexOperation.cs

https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Server/Documents/Handlers/IndexHandler.cs#L393

@mobilunity-user In general server should always return an object. If you spot a place, where it's not the case, please just let us know.

@IdanHaim Would you check DeleteIndex/DatabaseCommand in Python client please?